webpack/assets/javascripts/react_app/components/users/PersonalAccessTokens/PersonalAccessTokensReducer.js

```diff
@@ -0,0 +1,34 @@
+import Immutable from 'seamless-immutable';
+import {
+  PERSONAL_ACCESS_TOKENS_REQUEST,
+  PERSONAL_ACCESS_TOKENS_SUCCESS,
+  PERSONAL_ACCESS_TOKENS_FAILURE,
+  PERSONAL_ACCESS_TOKEN_FORM_SUBMITTED,
+  PERSONAL_ACCESS_TOKEN_CLEAR,
+} from './PersonalAccessTokensConstants';
+
+const initialState = Immutable({ tokens: [] });
+
+const sortTokens = tokens => tokens.sort((a, b) => a.created_at < b.created_at);
+
+export default (state = initialState, { type, payload }) => {
```

Comment:
You can use our API middleware which fetches API requests and store their values in a dedicated slot in the store.
 ```js
// dispatching
    dispatch(
      get({
        key: KEY,
        url: foremanUrl(`/path`),
      })
    );
// selecting
  const yourData = useSelector(state =>
    selectAPIResponse(state, KEY') // selects the data directly from the store without extra reducer
  );
```